optimum/exporters/tflite/convert.py

```diff
@@ -0,0 +1,183 @@
+        else:
+            logger.info(f"\t\t-[✓] {output.shape} matches {ref_output.shape}")
+
+        # Values
+        if not np.allclose(ref_output, output, atol=atol):
+            max_diff = np.amax(np.abs(ref_output - output))
+            logger.error(f"\t\t-[x] values not close enough, max diff: {max_diff} (atol: {atol})")
+            value_failures.append((name, max_diff))
+        else:
+            logger.info(f"\t\t-[✓] all values close (atol: {atol})")
+
+    if shape_failures:
+        msg = "\n".join(f"- {t[0]}: got {t[1]} (reference) and {t[2]} (TFLite)" for t in shape_failures)
+        raise ShapeError(
+            f"Output shapes do not match between reference model and TensorFlow Lite exported model:\n" "{msg}"
+        )
+
+    if value_failures:
+        msg = "\n".join(f"- {t[0]}: max diff = {t[1]}" for t in value_failures)
+        raise AtolError(
+            "The maximum absolute difference between the output of the reference model and the TensorFlow Lite "
+            f"exported model is not within the set tolerance {atol}:\n{msg}"
+        )
+
+
+def export(
+    model: "TFPreTrainedModel",
+    config: "TFLiteConfig",
+    output: Path,
+) -> Tuple[List[str], List[str]]:
+    """
+    Exports a TensorFlow model to a TensorFlow Lite model.
+
+    Args:
+        model ([`TFPreTrainedModel`]):
+            The model to export.
+        config ([`~exporters.tflite.TFLiteConfig`]):
+            The TFLite configuration associated with the exported model.
+        output (`Path`):
+            Directory to store the exported TFLite model.
+
+    Returns:
+        `Tuple[List[str], List[str]]`: A tuple with an ordered list of the model's inputs, and the named inputs from
+        the TFLite configuration.
+    """
+    if not is_tf_available():
+        raise ImportError("Cannot convert because TensorFlow is not installed. " "Please install TensorFlow first.")
+    import tensorflow as tf
+
+    output.parent.mkdir(parents=True, exist_ok=True)
+
+    logger.info(f"Using TensorFlow: {tf.__version__}")
+    model.config.return_dict = True
+
+    # Check if we need to override certain configuration item
+    if config.values_override is not None:
+        logger.info(f"Overriding {len(config.values_override)} configuration item(s)")
+        for override_config_key, override_config_value in config.values_override.items():
+            logger.info(f"\t- {override_config_key} -> {override_config_value}")
+            setattr(model.config, override_config_key, override_config_value)
+
+    signatures = config.model_to_signatures(model)
+
+    with TemporaryDirectory() as tmp_dir_name:
+        tf.saved_model.save(model, tmp_dir_name, signatures)
+        converter = tf.lite.TFLiteConverter.from_saved_model(tmp_dir_name)
+        tflite_model = converter.convert()
+
+    with open(output, "wb") as fp:
+        fp.write(tflite_model)
+
+    return config.inputs, config.outputs
```

Comment:
There is no optimization involved here. 

TFLite has three post-training optimization offerings through quantization and other things (kernel fusion, for example): https://www.tensorflow.org/lite/performance/post_training_quantization

Reply:
Yes, of course this is planned. The idea with this PR is to add the minimal code needed to iterate fast later. That is why there is no other configuration that the one for BERT.